service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/AmalgamatedLandingZoneService.java

```diff
@@ -0,0 +1,192 @@
+ */
+@Component
+public class AmalgamatedLandingZoneService implements WorkspaceLandingZoneService {
+
+  private final LandingApiClientTypeAdapter typeAdapter;
+  private final LandingZoneService landingZoneService;
+
+  @Autowired
+  public AmalgamatedLandingZoneService(LandingZoneService landingZoneService) {
+    this.landingZoneService = landingZoneService;
+    this.typeAdapter = new LandingApiClientTypeAdapter();
+  }
+
+  @Override
+  public ApiCreateLandingZoneResult startLandingZoneCreationJob(
+      BearerToken bearerToken,
+      String jobId,
+      UUID landingZoneId,
+      String definition,
+      String version,
+      List<ApiAzureLandingZoneParameter> parameters,
+      UUID billingProfileId,
+      String asyncResultEndpoint) {
+
+    LandingZoneRequest landingZoneRequest =
+        LandingZoneRequest.builder()
+            .landingZoneId(landingZoneId)
+            .definition(definition)
+            .version(version)
+            .parameters(MapperUtils.LandingZoneMapper.landingZoneParametersFrom(parameters))
+            .billingProfileId(billingProfileId)
+            .build();
+    var result =
+        landingZoneService.startLandingZoneCreationJob(
+            bearerToken, jobId, landingZoneRequest, asyncResultEndpoint);
+
+    return typeAdapter.toApiCreateLandingZoneResult(result);
+  }
+
+  @Override
+  public ApiDeleteAzureLandingZoneResult startLandingZoneDeletionJob(
+      BearerToken bearerToken, String jobId, UUID landingZoneId, String resultEndpoint) {
+    var result =
+        landingZoneService.startLandingZoneDeletionJob(
+            bearerToken, jobId, landingZoneId, resultEndpoint);
+    return typeAdapter.toApiDeleteAzureLandingZoneResult(result);
+  }
+
+  @Override
+  public ApiDeleteAzureLandingZoneJobResult getDeleteLandingZoneResult(
+      BearerToken bearerToken, UUID landingZoneId, String jobId) {
+    var response = landingZoneService.getAsyncDeletionJobResult(bearerToken, landingZoneId, jobId);
+    return typeAdapter.toApiDeleteAzureLandingZoneJobResult(response);
+  }
+
+  @Override
+  public ApiAzureLandingZoneResult getAsyncJobResult(BearerToken bearerToken, String jobId) {
+    var response = landingZoneService.getAsyncJobResult(bearerToken, jobId);
+    return typeAdapter.toApiAzureLandingZoneResult(response);
+  }
+
+  @Override
+  public ApiAzureLandingZone getAzureLandingZone(BearerToken bearerToken, UUID landingZoneId) {
+    LandingZone landingZoneRecord = landingZoneService.getLandingZone(bearerToken, landingZoneId);
+    return typeAdapter.toApiAzureLandingZone(landingZoneRecord);
+  }
+
+  @Override
+  public ApiAzureLandingZoneList listLandingZonesByBillingProfile(
+      BearerToken bearerToken, UUID billingProfileId) {
+    ApiAzureLandingZoneList result = new ApiAzureLandingZoneList();
+    List<LandingZone> landingZones =
+        landingZoneService.getLandingZonesByBillingProfile(bearerToken, billingProfileId);
+    if (!landingZones.isEmpty()) {
```

Comment:
Why enforce this here?
If LZS is returning a list, and this is returning a list, why not just do a straight map operation?
And if it _is_ enforced, why return a list at all?

Maybe the correct question to ask is "why is LZS returning a list here?"

Either way, I'm not sure it's correct for WSM to be enforcing LZS constraints.

Reply:
True. The LZS actually verifies this itself [here](https://github.com/DataBiosphere/terra-landing-zone-service/blob/ceafbf7d5bf25554dafbc25c1f4f282edd8eba4b/service/src/main/java/bio/terra/lz/futureservice/app/service/LandingZoneAppService.java#L67), so this logic is redundant.  Removed.